install.py

```diff
@@ -44,21 +45,29 @@ def pip_install_requirements(requirements_txt="requirements.txt"):
         sys.exit(-1)
     print("OK")
 
-    if args.component == "distributed":
-        success, errmsg = pip_install_requirements(requirements_txt="torchbenchmark/util/distributed/requirements.txt")
+    if args.setup_only:
+        if args.component:
+            print("--component and --setup-only cannot be specified together.")
+            sys.exit(-1)
+        print("Requirements installation will be skipped")
+        success = True
+    else:
+        if args.component == "distributed":
+            success, errmsg = pip_install_requirements(requirements_txt="torchbenchmark/util/distributed/requirements.txt")
+            if not success:
+                print("Failed to install torchbenchmark distributed requirements:")
+                print(errmsg)
+                if not args.continue_on_fail:
+                    sys.exit(-1)
+            sys.exit(0)
+
+        success, errmsg = pip_install_requirements()
         if not success:
-            print("Failed to install torchbenchmark distributed requirements:")
+            print("Failed to install torchbenchmark requirements:")
             print(errmsg)
             if not args.continue_on_fail:
                 sys.exit(-1)
-        sys.exit(0)
 
-    success, errmsg = pip_install_requirements()
-    if not success:
-        print("Failed to install torchbenchmark requirements:")
-        print(errmsg)
-        if not args.continue_on_fail:
-            sys.exit(-1)
     from torchbenchmark import setup
     success &= setup(models=args.models, verbose=args.verbose, continue_on_fail=args.continue_on_fail, test_mode=args.test_mode, allow_canary=args.canary)
```

Comment:
I think this is incorrect - even when running with `--setup-only`, the `setup()` function will still install all the requirements.txt files in each models directory, such as https://github.com/pytorch/benchmark/blob/main/torchbenchmark/util/framework/detectron2/requirements.txt and https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/drq/requirements.txt

Reply:
where/when these extra requirements.txt are installed? They seem to be harmless as they are just a few and not pinning any particular version (which is the issue we want to prevent). I will double check whether we can improve the `--setup-only` by preventing these extra requirements too

Reply:
> where/when these extra requirements.txt are installed? They seem to be harmless as they are just a few and not pinning any particular version (which is the issue we want to prevent). I will double check whether we can improve the `--setup-only` by preventing these extra requirements too

Unfortunately, there are plenty of them and many of them are pinning to particular versions. Almost every model comes with its own `requirements.txt` file, to name a few:
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/BERT_pytorch/requirements.txt
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/soft_actor_critic/requirements.txt
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/tacotron2/requirements.txt#L6
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/fastNLP_Bert/requirements.txt
https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/doctr_det_predictor/requirements.txt

The reason we have to pin the versions of the dependencies is because some model code are old and there's no update for ~1year, therefore, they won't work on the latest dependency code and it is non-trivial to migrate. However, our main customer (PyTorch team) still want to keep those and test against them.

Reply:
I am running a full test to check whether the "default models"(when we don't set `--models`) break an existing env

Do you know where I can find the code that would follow and install the `requirements.txt` file above? maybe this PR can go a step further and skip the installation of the model-specific requirements

Reply:
Yes that is another option, the code is at https://github.com/pytorch/benchmark/blob/main/torchbenchmark/__init__.py#L149